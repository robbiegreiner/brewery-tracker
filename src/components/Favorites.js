// import firebase from '../firebase.js';
import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Favorites extends Component {
  constructor() {
    super();
    this.state = {
    };
  }

  componentDidMount() {
    this.props.getFavorites(this.props.user.id);
  }

  render() {
    return (
      <div className='favorites'>
        <h1>Favorites Here!</h1>
      </div>
    );
  }

}

Favorites.propTypes = {
  user: PropTypes.object,
  getFavorites: PropTypes.func
};

export default Favorites;
