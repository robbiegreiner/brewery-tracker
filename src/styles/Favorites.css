.favorites {
  margin-top: 120px; }
