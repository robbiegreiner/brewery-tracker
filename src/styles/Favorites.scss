.favorites {
  margin-top: 120px;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center
}
