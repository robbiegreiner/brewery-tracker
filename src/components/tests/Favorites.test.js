import React from "react";
import Favorites from '../Favorites';
import { shallow } from "enzyme";

describe('Favorites component', () => {
  const mockFn = jest.fn();
  const wrapper = shallow(<Favorites
    user={{}}
    favorites={{}}
    favoriteBeers={[]}
    favoriteBreweries={[]}
    getFavorites={mockFn}
    getFavoriteBeers={mockFn}
    getFavoriteBreweries={mockFn}
    setCurrentBeer={mockFn}
    removeFavoriteBeer={mockFn}
    removeFavoriteBreweries={mockFn}
    addFavoriteBeer={mockFn}
    addFavoriteBrewery={mockFn}/>);


  it('should match snapshot', () => {
    // console.log(wrapper.debug());
    expect(wrapper).toMatchSnapshot();
  });
});
