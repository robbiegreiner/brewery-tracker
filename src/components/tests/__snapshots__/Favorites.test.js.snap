// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    favoriteBeers={Array []}
    favoriteBreweries={Array []}
    favorites={Object {}}
    getFavoriteBeers={[Function]}
    getFavoriteBreweries={[Function]}
    removeFavoriteBeer={[Function]}
    removeFavoriteBreweries={[Function]}
    setCurrentBeer={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Your Favorites Are Empty!
</h1>,
      "className": "favorites",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "Your Favorites Are Empty!",
      },
      "ref": null,
      "rendered": "Your Favorites Are Empty!",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Your Favorites Are Empty!
</h1>,
        "className": "favorites",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Your Favorites Are Empty!",
        },
        "ref": null,
        "rendered": "Your Favorites Are Empty!",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
